Tests/Foundation/Tests/TestFileManager.swift

```diff
@@ -1755,6 +1757,49 @@ VIDEOS=StopgapVideos
         }
         #endif
     }
+
+    /**
+     Tests that we can get an item replacement directory concurrently.
+
+     - Bug: [SR-12272](https://bugs.swift.org/browse/SR-12272)
+     */
+    func test_concurrentGetItemReplacementDirectory() throws {
+        let fileManager = FileManager.default
+
+        let operationCount = 10
+
+        var directoryURLs = [URL?](repeating: nil, count: operationCount)
+        var errors = [Error?](repeating: nil, count: operationCount)
+
+        let dispatchGroup = DispatchGroup()
+        for operationIndex in 0..<operationCount {
+            DispatchQueue.global().async(group: dispatchGroup) {
+                do {
+                    let directory = try fileManager.url(for: .itemReplacementDirectory,
+                                                        in: .userDomainMask,
+                                                        appropriateFor: URL(fileURLWithPath: NSTemporaryDirectory(),
+                                                                            isDirectory: true),
+                                                        create: true)
+                    directoryURLs[operationIndex] = directory
+                } catch {
+                    errors[operationIndex] = error
+                }
+            }
+        }
+        dispatchGroup.wait()
+
+        for directoryURL in directoryURLs {
+            if let directoryURL = directoryURL {
+                try? fileManager.removeItem(at: directoryURL)
+            }
+        }
+
+        for error in errors {
+            if let error = error {
+                throw error
```

Comment:
Probably better to use `XCTFail(error)` here to show the error.

Reply:
Why?

Reply:
Ping. Can you elaborate on why `XCTFail` is preferable to throwing the error?

Reply:
For functions that can fail when setting up the test, it is appropriate to just throw to exit the exit the entire test as nothing else would be expected to work. However for the main part of the test an `XCT...` allows a more useful output string and shows other people reading the test what is actually being tested.

A good example is `URL.init?(string:)`. This returns an `Optional` but would not be expected to fail if given, say, `http://www.example.com`. So if a `URL` needs to be created at the start of the test then:
```
let url = try XCTUnwrap(URL(string: "http://example.com/"))
```
which will `throw` is fine as something has gone seriously wrong if this fails and the rest of the test is going to break.

However if we are writing tests for `URL.init?(string:)` itself, then the following would be used:
```XCTAssertNotNil(URL(string: "http://example.com/"))```

As that is the function actually being tested.
 Note this isnt an exact rule as sometimes its easier to `throw` when testing certain parts  but in your test above the errors are being caught anyway so using an `XCTFail()` is no extra work.